OpenRA.Mods.D2k/SpriteLoaders/R816Loader.cs

```diff
@@ -59,7 +64,21 @@ public R8Frame(Stream s)
 				// Skip alignment byte
 				s.ReadUInt8();
 
-				Data = s.ReadBytes(width * height);
+				if (bpp == 8)
+					Data = s.ReadBytes(width * height);
+				else
+				{
+					Data = new byte[width * height * 3];
+
+					for (var i = 0; i < Data.Length;)
+					{
+						var color16 = s.ReadUInt16();
+
+						Data[i++] = (byte)(((color16 >> 7) & 0xf8) | ((color16 >> 12) & 0x07));
```

Comment:
Optional. Add some comment explaining the magic. May help future developers.

`0xf8=11111000`.  

File format is bgr and you store rgb?